### PR TITLE
HV Reset

### PR DESCRIPTION
**Pull request by OCopping**
Add HV Reset routine

### CODE REVIEW

File: python/src/eiger_detector/control/eiger_detector.py [hunk 1]
```diff
@@ -854,6 +855,32 @@ def lv_loop(self):
                 self.read_detector_live_image()
             time.sleep(0.1)
 
+    def hv_reset(self):
```
[the remark this answers is not in the document]
Comment (GDYendell) [reply python/src/eiger_detector/control/eiger_detector.py:858]: What I mean is, if we merge this as is there is no way to run this function via an HTTP request.

I think it just needs an entry adding to the `get` and `set` methods, like `stop_acquisition` has.

File: python/src/eiger_detector/control/eiger_detector.py [hunk 2]
```diff
@@ -854,6 +855,36 @@ def lv_loop(self):
...
+        logging.info("Initiating HV Reset")
+
+        material = getattr(self, "sensor_material")
+        # Make sure sensor material is CdTe
+        try:
+            assert material.lower() == "cdte"
+        except AssertionError:
+            logging.info("Sensor material is not CdTe.")
+            return
```
Comment (GDYendell) [inline python/src/eiger_detector/control/eiger_detector.py:867]: I meant something more like
```suggestion
       if material.lower() != "cdte"
           logging.error("Sensor material is not CdTe - Not running HV reset")
           return
```

and also replace the `raise Exception` below with something similar.

File: python/src/eiger_detector/control/eiger_detector.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GDYendell) [reply python/src/eiger_detector/control/eiger_detector.py:858]: Or better yet, like initialising, which reports a state to show it is currently running. The HV reset runs for long time so it should have an LED to show it is running.

File: python/src/eiger_detector/control/eiger_detector.py [hunk 3]
```diff
@@ -854,6 +877,35 @@ def lv_loop(self):
                 self.read_detector_live_image()
             time.sleep(0.1)
 
+    def hv_reset(self):
+        logging.info("Initiating HV Reset")
+
+        material = getattr(self, "sensor_material")
+        # Make sure sensor material is CdTe
+        if material.lower() != "cdte":
+            logging.info("Sensor material is not CdTe.")
```
Comment (GDYendell) [inline python/src/eiger_detector/control/eiger_detector.py:886]: I think both messages should be error, not info

Comment (GDYendell) [reply python/src/eiger_detector/control/eiger_detector.py:886]: and this one should say that it is not going to run the reset.

File: python/src/eiger_detector/control/eiger_detector.py [hunk 4]
```diff
@@ -854,6 +877,35 @@ def lv_loop(self):
...
+
+        # send HV reset command, 45 seconds is recommended, in a loop of 10
+        niterations = 10
+        for _ in range(niterations):
+            counter = 0
+            while hv_state != "READY":
+                counter += 1
+                if counter > 60:
+                    logging.info("Detector failed to be ready after 600 seconds,\
+                                     stopping hv reset")
```
Comment (GDYendell) [inline python/src/eiger_detector/control/eiger_detector.py:900]: Formatting
```suggestion
                    logging.error(
                        "Detector failed to be ready after 600 seconds, "
                        "stopping hv reset"
                    )
```

**Comment by GDYendell** [comment]
TODO:
- [ ] Squash into one commit
- [ ] Test on I19

**Comment by OCopping** [comment]
@GDYendell are you happy for me to merge this?
Apart from the rare warning the routine appears to run fine (I can look into this warning in the future)

**Pull request merged**